shell/embed.ml

```diff
@@ -0,0 +1,22 @@
+let add_chan buf chan =
+  try
+    while true do
+      Buffer.add_char buf (input_char chan);
+    done
+  with End_of_file ->
+    ()
+
+let () =
+  let setup = Sys.argv.(1) in
+  let content_setup =
+    let ic = open_in_bin setup in
+    let bs = Buffer.create 3_000_000 in
+    add_chan bs ic;
+    close_in ic;
+    Buffer.contents bs
+  in
+  Printf.printf
+    {|(* THIS FILE IS AUTOMATICALLY GENERATED, EDIT configure.ac & shell/embed.ml INSTEAD *)
+let content = Some "%s"
+|}
+    (String.escaped content_setup)
```

Comment:
We loose some compression by not using base64 (2.1M vs 1.5M)

Reply:
That's not what i'm observing. Base64 doesn't compress; on the contrary as far as i know it makes the data bigger due to its encoding.
Test done encoding my local `/usr/bin/cat` ELF binary:
- base64: 1.87MB
- String.escaped: 1.77MB

Reply:
Erm, apples and oranges are being compared here - the base64 version was writing an actual base64 string into the _executable_ - so the setup executable will be 33% larger. This version is writing the OCaml string constant to the _source file_ ... so the data in the executable will be a verbatim copy of the original executable (plus a tiny bit of overhead from the marshal header)

Reply:
Indeed, I tested the `base64` tool and diffed the generated file, not the library directly in the code.